mlx/coverity_services.py

```diff
@@ -1,614 +1,396 @@
+                    {
+                        "class": "Stream",
+                        "name": stream,
+                        "type": "nameMatcher"
+                    }
+                ]
+            }
+        ]
+
+        Filter = namedtuple("Filter", "name matcher_type list allow_regex", defaults=[None, False])
+        filter_options = {
+            "checker": Filter("Checker", "keyMatcher", self.checkers, True),
+            "impact": Filter("Impact", "keyMatcher", IMPACT_LIST),
+            "kind": Filter("Issue Kind", "keyMatcher", KIND_LIST),
+            "classification": Filter("Classification", "keyMatcher", CLASSIFICATION_LIST),
+            "action": Filter("Action", "keyMatcher", ACTION_LIST),
+            "cwe": Filter("CWE", "idMatcher"),
+            "cid": Filter("CID", "idMatcher")
+        }
+
+        for option, filter in filter_options.items():
+            if filters[option]:
+                filter_values = self.handle_attribute_filter(
+                    filters[option], filter.name, filter.list, filter.allow_regex
+                )
+                if filter_values:
+                    query_filters.append(self.assemble_query_filter(filter.name, filter_values, filter.matcher_type))
+
+        if filters["component"]:
+            filter_values = self.handle_component_filter(filters["component"])
+            if filter_values:
+                query_filters.append(self.assemble_query_filter("Component", filter_values, "nameMatcher"))
+
+        data = {
+            "filters": query_filters,
+            "columns": list(self.column_keys(column_names)),
+            "snapshotScope": {
+                "show": {
+                    "scope": "last()",
+                    "includeOutdatedSnapshots": False
+                },
+                "compareTo": {
+                    "scope": "last()",
+                    "includeOutdatedSnapshots": False
+                }
+            }
+        }
+
+        logging.info("Running Coverity query...")
+        return self.retrieve_issues(data)
 
     def handle_attribute_filter(self, attribute_values, name, *args, **kwargs):
-        """ Applies any filter on an attribute's values.
+        """Applies any filter on an attribute's values.
 
         Args:
             attribute_values (str): A CSV list of attribute values to query.
             name (str): String representation of the attribute.
+
+        Returns:
+            list[str]: The list of valid attributes
         """
-        logging.info('Using %s filter [%s]', name, attribute_values)
-        validated = self.config_service.add_filter_rqt(name, attribute_values, *args, **kwargs)
-        logging.info('Resolves to [%s]', validated)
-        if validated:
-            self.filters += ("<%s(%s)> " % (name, validated))
+        logging.info("Using %s filter [%s]", name, attribute_values)
+        filter_values = self.validate_filter_option(name, attribute_values, *args, **kwargs)
```

Comment:
Please remove `validate_filter_option` and move its contents here. This `handle_attribute_filter` is just a wrapper now that logs a duplicate INFO.

Reply:
Resolved in 2c012cde5a99ceba0d87196f87a0514f97ccba74